### PR TITLE
Add a dynamic ASG test to the security_groups test suite

### PR DESCRIPTION
**Pull request by reneighbor**
### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

Yes

### What is this change about?

Acceptance tests for Dynamic ASGs, to be enabled by default in cf-deployment.

### Please provide contextual information.

https://www.pivotaltracker.com/story/show/181363273

### What version of cf-deployment have you run this cf-acceptance-test change against?

v18.0.0 with silk-release update to v 3.0 and policy-server-asg-syncer enabled with this ops file: https://www.pivotaltracker.com/story/show/180837394/comments/229193630

### Please check all that apply for this PR:
- [x] introduces a new test --- Are you sure everyone should be running this test?
- [ ] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?
- [ ] YES
- [x] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

xCATs should validate common operator workflows._
_CATs is not a regression test suite._
_CATs is run by every component team to validate their releases before promotion._

### How should this change be described in cf-acceptance-tests release notes?

Add a test for dynamic (no restart) security groups.

### How many more (or fewer) seconds of runtime will this change introduce to CATs?
3 mins

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### CODE REVIEW

File: assets/proxy/main.go [hunk 1]
```diff
@@ -93,5 +94,8 @@ var httpClient = &http.Client{
 			Timeout:   10 * time.Second,
 			KeepAlive: 0,
 		}).Dial,
+		TLSClientConfig: &tls.Config{
```
Comment (davewalter) [inline assets/proxy/main.go:97]: Was this change required, or was this added to work around a limitation with your test environment?

**Review by davewalter — COMMENTED** [review]
Hi @reneighbor,

Since we merged https://github.com/cloudfoundry/cf-deployment/pull/957 and bumped cf-networking & silk to v3.3.0, we have started seeing failures in two specific tests:

* [security_groups] App Instance Networking Using container-networking and running security-groups [It] correctly configures asgs and c2c policy independent of each other ([ security_groups/running_security_groups.go:237](https://github.com/cloudfoundry/cf-acceptance-tests/blob/7f63f94c7b6d96771219a6af5af7fcb737cddae8/security_groups/running_security_groups.go#L237))

* [tasks] v3 tasks when associating a task with an app and binding a space-specific ASG [It] applies the associated app's ASGs to the task  ([tasks/task.go:354](https://github.com/cloudfoundry/cf-acceptance-tests/blob/7f63f94c7b6d96771219a6af5af7fcb737cddae8/tasks/task.go#L354))

You can see examples of the failures in CI [here](https://release-integration.ci.cf-app.com/teams/main/pipelines/cf-deployment/jobs/fresh-cats/builds/3674), [here](https://release-integration.ci.cf-app.com/teams/main/pipelines/cf-deployment/jobs/upgrade-cats/builds/3423) and [here](https://release-integration.ci.cf-app.com/teams/main/pipelines/cf-deployment/jobs/experimental-cats/builds/2363).

Do we need to make adjustments to these two tests as well for compatibility with dynamic ASGs?

Dave

**Review by ctlong — APPROVED** [review]
LGTM

**Comment by risicle** [comment]
This seems to have made its way onto the `cf19.0` branch, which we assumed could be relied upon by people running vanilla cf-deployment 19.0.0.

**Comment by ctlong** [comment]
@risicle you're correct. That merge happened in error, so I've reset the cf19.0 and release-candidate branches back to the commit before this was merged.
